flake8_trio.py

```diff
@@ -200,6 +206,26 @@ def has_decorator(decorator_list: List[ast.expr], *names: str):
     return False
 
 
+# matches the full decorator name against fnmatch pattern
+def fnmatch_decorator(decorator_list: List[ast.expr], *patterns: str):
+    def construct_name(expr: ast.expr) -> str:
+        if isinstance(expr, ast.Name):
+            return expr.id
+        assert isinstance(expr, ast.Attribute)
+        return construct_name(expr.value) + "." + expr.attr
+
+    for decorator in decorator_list:
+        qualified_decorator_name = construct_name(decorator)
+        for pattern in patterns:
+            # remove any leading @
```

Comment:
`str.lstrip()` is great for this. I'd also use `any()` for the inner loop

Reply:
`lstrip` def cleaner :+1:

I don't think `any()` is cleaner here
```python
for pattern in patterns:
    if fnmatch(qualified_decorator_name, pattern.lstrip("@")):
        return True
```
vs
```python
if any(
    fnmatch(qualified_decorator_name, pattern.lstrip("@"))
    for pattern in patterns
):
    return True
```
and they should be the same efficiency